In [1]:
import numpy as np
from scipy.optimize import linprog
c = [-2, -3, 0, 1, 0, 0]  
A_eq = [
    [2, -1, 0, -2, 1, 0],
    [3,  2, 1, -3, 0, 0],
    [-1, 3, 0, 4, 0, 1]
]
b_eq = [16, 18, 24]
x_bounds = [(0, None) for _ in range(6)]

res = linprog(c, A_eq=A_eq, b_eq=b_eq, bounds=x_bounds, method='highs')

res


        message: Optimization terminated successfully. (HiGHS Status 7: Optimal)
        success: True
         status: 0
            fun: -25.636363636363637
              x: [ 5.455e-01  8.182e+00  0.000e+00  0.000e+00  2.309e+01
                   0.000e+00]
            nit: 3
          lower:  residual: [ 5.455e-01  8.182e+00  0.000e+00  0.000e+00
                              2.309e+01  0.000e+00]
                 marginals: [ 0.000e+00  0.000e+00  8.182e-01  3.636e-01
                              0.000e+00  4.545e-01]
          upper:  residual: [       inf        inf        inf        inf
                                    inf        inf]
                 marginals: [ 0.000e+00  0.000e+00  0.000e+00  0.000e+00
                              0.000e+00  0.000e+00]
          eqlin:  residual: [ 0.000e+00  0.000e+00  0.000e+00]
                 marginals: [-0.000e+00 -8.182e-01 -4.545e-01]
        ineqlin:  residual: []
                 marginals: []
 mip_node_count: 0
 mip_dual_bo

Так как linprog решает задачу минимизации, то для максимизации результат получается с обратным знаком L (max) = 25.64. Оптимальные значения переменных x1=0.545, x2=8.18, x3=x4=x6=0, x5=23.09. Данные значения совпадают с аналитическим результатом